base/irrationals.jl

```diff
@@ -221,6 +221,15 @@ end
 log(::Irrational{:e}) = 1 # use 1 to correctly promote expressions like log(x)/log(e)
 log(::Irrational{:e}, x::Number) = log(x)
 
+# These methods should go to special/trig.jl, but Irrational is not yet defined when the
+# file is loaded.
+sinpi(x::Irrational) = sinpi(float(x))
+cospi(x::Irrational) = cospi(float(x))
+sinc(x::Irrational) = sinc(float(x))
+cosc(x::Irrational) = cosc(float(x))
+sinc(z::Complex{<:Irrational}) = sinc(float(z))
+cosc(z::Complex{<:Irrational}) = cosc(float(z))
```

Comment:
See, now this is a bit of a slippery slope. When do we stop implementing functions for Irrationals? Why these functions and not others?